src/Arr.php

```diff
@@ -133,4 +133,14 @@ public static function assocIn(array $array, array $path, $value)
     {
         return static::updateIn($array, $path, fn () => $value);
     }
+
+    /**
+     * @param mixed[] $array
+     *
+     * @return mixed[]
+     */
+    public static function removeAndReIndex(array $array, callable $fn): array
```

Comment:
We should just call this `filter`. `Arr::filter`'s claim to fame will be that it doesn't return sparse keys 💯

Reply:
I was struggling to come up with a good name for this... I like that idea... our helper does the magic! :handshake: